Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,83 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
         {
             var segmentStatements = new List<string>();
             var parameters = new List<DbParameter>();
+            var unsupportedAuthorizationViews = new List<string>();
+            var edOrgTypes = new List<string>();
 
             foreach (var authorizationSegment in authorizationSegments)
             {
-                // Within each claim segment, group the values by ed org type, and combine with "OR"
-                var claimEndpointsGroupedByName = authorizationSegment.ClaimsEndpoints
-                    .GroupBy(ep => ep.Name)
-                    .Select(g => g)
-                    .ToList();
+                string claimEndpointName = authorizationSegment.ClaimsEndpoints.FirstOrDefault().Name;
 
-                var segmentExpressions = new List<string>();
+                if (EducationOrganizationEntitySpecification.IsEducationOrganizationIdentifier(claimEndpointName))
+                {
+                    claimEndpointName = "EducationOrganizationId";
+                }
 
-                var unsupportedAuthorizationViews = new List<string>();
+                string subjectEndpointName = authorizationSegment.SubjectEndpoint.Name;
```

Comment:
Continuing with my previous comment on immutability... please, as much as possible, try to assign variables once. Rewrite this block as follows:

```
string subjectEndpointName = EducationOrganizationEntitySpecification.IsEducationOrganizationIdentifier(authorizationSegment.SubjectEndpoint.Name)
    ? "EducationOrganizationId"
    : authorizationSegment.SubjectEndpoint.Name;
```
And once it is set... leave the value alone.

Reply:
I updated it